### PR TITLE
Add option to run bbcli against local app.

### PR DESCRIPTION
**Pull request by vadimberezniker**
Instead of a single `--dev` flag there's now a `--env` flag that accepts
one of prod|dev|local.

There are some other small changes in this PR:
 - Moved networking utilities out of enterprise directory to avoid bbcli
-> enterprise dependency.
 - `FindWorkspaceFile` moved to `bazel` package to be reused for runfile
changes.
 - Using `--remote_download_toplevel` for remote bazel builds in order
to support running.

---

**Version bump**: None

### CODE REVIEW

File: cli/autoconfig/autoconfig.go [hunk 1]
```diff
@@ -57,41 +60,77 @@ func remoteBazelArgs(besBackend, remoteExecutor string) []string {
...
+	for _, a := range addrs {
+		if v, ok := a.(*net.IPNet); ok {
+			return v.IP, nil
+		}
 	}
+	return nil, fmt.Errorf("could not determine host's default IP")
+}
 
-	wsFilePath, err := findWorkspaceFile()
+// TODO(vadim): reduce # of return values
```
Comment (tylerwilliams) [inline cli/autoconfig/autoconfig.go:80]: lol sometimes structs are good, actually :P

File: cli/autoconfig/autoconfig.go [hunk 2]
```diff
@@ -57,41 +60,77 @@ func remoteBazelArgs(besBackend, remoteExecutor string) []string {
...
+func Configure(ctx context.Context, bazelFlags *commandline.BazelFlags, filteredOSArgs []string) (*commandline.BazelFlags, *BazelOpts, []string, error) {
+	grpcEndpoint := ""
+	webURL := ""
```
Comment (tylerwilliams) [inline cli/autoconfig/autoconfig.go:83]: nit: 

var grpcEndpoint, webURL string

File: cli/autoconfig/autoconfig.go [hunk 3]
```diff
@@ -57,41 +60,77 @@ func remoteBazelArgs(besBackend, remoteExecutor string) []string {
...
+			filteredOSArgs = append(filteredOSArgs, "--remote_download_toplevel")
 		}
-		filteredOSArgs = append(filteredOSArgs, "--remote_download_minimal")
 		filteredOSArgs = append(filteredOSArgs, "--jobs=200")
 	}
 
 	if (*cache || *remote == remoteBazel) && bazelFlags.RemoteCache == "" {
-		bazelFlags.RemoteCache = fmt.Sprintf("grpcs://remote.%s", serviceDomain)
+		bazelFlags.RemoteCache = grpcEndpoint
 	}
```
Comment (tylerwilliams) [inline cli/autoconfig/autoconfig.go:130]: <not this cl> but it seems like there's a lot going on here

**Review by tylerwilliams — APPROVED** [review]

File: cli/autoconfig/autoconfig.go (hunk 2 above)
Comment (vadimberezniker) [reply cli/autoconfig/autoconfig.go]: Done.

File: cli/autoconfig/autoconfig.go (hunk 3 above)
Comment (vadimberezniker) [reply cli/autoconfig/autoconfig.go:129]: I agree. (this whole func can use some love, i will take a look after i'm done with the run changes)

**Pull request merged**